imports/plugins/core/core/server/method-hooks.js

```diff
@@ -74,52 +76,56 @@ MethodHooks._initializeHook = function (mapping, methodName, hookFunction) {
     // Get arguments you can mutate
     const args = _.toArray(inputArgs);
     let beforeResult;
-    // Call the before hooks
+    let hooksProcessed = 0;
 
-    const beforeHooks = MethodHooks._beforeHooks[methodName];
-    _.each(beforeHooks, (beforeHook, hooksProcessed) => {
-      beforeResult = beforeHook.call(this, {
+    // Call the before hooks
+    const beforeHooks = MethodHooks._beforeHooks[methodName] || [];
+    for (const beforeHook of beforeHooks) {
+      beforeResult = Promise.await(beforeHook.call(this, {
         result: undefined,
         error: undefined,
         arguments: args,
         hooksProcessed
-      });
+      }));
+
+      hooksProcessed += 1;
 
       if (beforeResult === false) {
+        check(args, [Match.Any]);
```

Comment:
Let's move this to the top of the method. 
Since we are kind of skipping the check. It makes more sense to move this to the top.

Reply:
If we do that, then methods that don't check all their args won't be able to take advantage of `audit-arguments-checks`. If we have it here, then we basically skip `audit-arguments-checks` only when it's not needed - i.e., exiting early, before the main method.

Reply:
Yes makes sense.